### PR TITLE
Feat/support nvidia docker and nonroot

### PR DESCRIPTION
**Pull request by VinceCui**
### Describe what this PR does / why we need it
This PR support these features:
1. Support non root user use sudo to install.
2. If node to scale is alreay in current k8s cluster, just skip it.
3. Split NewClusterFile to NewClusterFile and GetActualClusterFile.
4. Support get customize cri socket path for each node.
5. Add upgrade-host plugin.

### CODE REVIEW

**Comment by cubxxw** [comment]
#### resolve

Resolved issue: #1491、#1999

**Resolved remote read permission to `sshd_config` file:**
```bash
$ ls -al /etc/ssh/sshd_config
-rw------- 1 root root 3797 Feb 15 16:35 /etc/ssh/sshd_config
$ sudo chmod 644 /etc/ssh/sshd_config
```

<br>

#### existing problem

+ [ ] **Still needs to assign non-secret permissions to user sudo(Every node except master0):**
```bash
# echo "sealer ALL=(ALL) NOPASSWD: NOPASSWD: ALL" >> /etc/sudoers
```

+ [ ] **The rootless feature of [buildah](https://github.com/containers/buildah) is not addressed:**

**Review by cubxxw — APPROVED** [review]

**Review by cubxxw — APPROVED** [review]

File: cmd/sealer/cmd/cluster/scale-up.go [hunk 1]
```diff
@@ -160,12 +161,6 @@ func scaleUpCluster(clusterImageName string, scaleUpMasterIPList, scaleUpNodeIPL
 		return err
 	}
 
-	//we need to save desired clusterfile to local disk temporarily.
-	//and will use it later to clean the cluster node if ScaleUp failed.
```
Comment (Starnop) [inline cmd/sealer/cmd/cluster/scale-up.go:164]: it looks like we should not delete the code according to the comments

File: pkg/runtime/kubernetes/utils.go [hunk 2]
```diff
@@ -53,6 +53,7 @@ const (
 )
 
 const (
+	GetCustomizeCRISocket   = "cat /etc/sealer-cri/socket-path"
```
Comment (Starnop) [inline pkg/runtime/kubernetes/utils.go:56]: /etc/sealerio/cri/socket-path

**Review by cubxxw — APPROVED** [review]

**Review by kakaZhou719 — APPROVED** [review]
LGTM

**Pull request merged**